File: src/LectureID.java

public class LectureID implements UniqueID {
	
	private String departmentCode;
	private int yearCode;
	private int lectureCode;
	
	public void setID(String string) {
		
	}
	
	public void setID(String DepartmentCode, int YearCode, int LectureCode) {
		this.departmentCode = DepartmentCode;
		this.yearCode = YearCode;
		this.lectureCode = LectureCode;
	}
	
	public String getID() {
		return departmentCode +""+ yearCode +""+ lectureCode;
	}

}

File: CSE3063F22P1_GRP1_iteration3/pythonCode/LectureID.py
from UniqueID import UniqueID

class LectureID(UniqueID):
    def __init__(self, __lectureCode: str):
        self.__lectureCode = __lectureCode

     # Creating properties for variables

    def setID(self, string: str):
        self.__lectureCode = string

    def getID(self) -> str:
        return self.__lectureCode
